documentation/puppet-api/v4/catalog.markdown

```diff
@@ -0,0 +1,76 @@
+---
+layout: default
+title: "Puppet Server: Puppet API: Catalog"
+canonical: "/puppetserver/latest/puppet-api/v4/catalog.html"
+---
+
+The catalog API returns a compiled catalog for the node specified in the request,
+making use of provided metadata like facts or environment if specified.
+If not specified, it will attempt to fetch this data from Puppet's configured sources
+(usually PuppetDB or a node classifier). The returned catalog is in JSON format,
+ready to be parsed and applied by an agent.
+
+## `POST /puppet/v4/catalog`
+
+(Introduced in Puppet Server 6.3.0)
+
+The input data for the catalog to be compiled is submitted as a JSON body with the 
+following form:
+```
+{
+  "certname": "<node name>",
+  "persistence": { "facts": <true/false>, "catalog": <true/false> },
+  # The rest are optional:
+  "facts": { "values": { "<fact name>": <fact value>, ... } },
+  "trusted_facts": { "values": { "<fact name>": <fact value>, ... } },
+  "environment": "<environment name>",
+  "options": { "prefer_requested_environment": <true/false>,
+               "capture_logs": <true/false> }
+}
+```
+
+#### `certname` (required)
+The name of the node for which to compile the catalog.
+
+#### `persistence` (required)
+A hash containing two required keys, `facts` and `catalog`, which when set to true will
+cause the facts and reports to be stored in PuppetDB, or discarded if set to false.
+
+#### `facts`
+A hash with a required `values` key, containing a hash of all the facts for the node.
+If not provided, Puppet will attempt to fetch facts for the node from PuppetDB.
+
+#### `trusted_facts`
```

Comment:
Do we want to call this `trusted_facts`? Puppetdb I think just uses `trusted` and internally to puppet it's always `trusted_data`. I don't know how we communicate this generally outside the company.

Reply:
That's a good question. I think most of our external discussion of trusted facts calls them that, "trusted facts". And that is how they were referenced by the people who wrote the tickets for this endpoint. I personally consider PDB's and Puppet's internal names confusing implementation details.